core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -59,7 +87,17 @@ oppia.factory('explorationData', [
             $log.info('Retrieved exploration data.');
             $log.info(response.data);
 
+            // Initialize the changeList by the one received from the backend.
             explorationData.data = response.data;
+            changeListService.loadAutosavedChangeList(
```

Comment:
Oh, I remember leaving a comment here, but looks like the network connection failed me. So I'll do that again :)

@seanlip -  Thanks. I've followed your suggestion and now the errors of the circular dependencies have been resolved!